test/uk/gov/hmrc/channelpreferences/controllers/PreferenceControllerSpec.scala

```diff
@@ -83,21 +88,176 @@ class PreferenceControllerSpec extends PlaySpec with ScalaFutures with MockitoSu
+      val fakePostRequest = FakeRequest("POST", "", Headers("Content-Type" -> "application/json"), postData)
+      val response = controller.confirm().apply(fakePostRequest)
+      status(response) mustBe UNAUTHORIZED
+      contentAsString(response) mustBe """{"reason":"SAUTR in Auth token is different from SAUTR in entity resolver"}"""
+      verify(mockEntityResolverConnector, never()).update(any[Entity]())(any[HeaderCarrier]())
+    }
+
+    // Case 1.3
+    """return UNAUTHORIZED (401) when SAUTR in authToken is linked to a different entityId in entity-resolver""" in new TestSetup {
+      pending
+      // WARNING: this scenario seems already included in the previous one
     }
 
-    """return CONFLICT (409) for a "magic" entityId""" in new TestSetup {
-      val postData: JsValue =
-        Json.parse(s"""{"entityId": "450262a0-1842-4885-8fa1-6fbc2aeb867d","itsaId": "itsa-id"}""")
+    // Case 1.4
+    """return OK (200) upon linking the given itsaId when SAUTR does not exist in the Auth token""" in new TestSetup {
+      reset(mockEntityResolverConnector)
+      reset(mockAuthConnector)
+
+      when(
+        mockAuthConnector.authorise[Option[String]](any[Predicate](), any[Retrieval[Option[String]]]())(
+          any[HeaderCarrier](),
+          any[ExecutionContext]()))
+        .thenReturn(Future.successful(None)) // Couldn't retrieve SAUTR from the authToken!
+
+      val passedBackEntityId = "passedBackEntityId"
+      val passedBackItsaId = "passedBackItsaId"
+
+      val resolvedEntity_saUtr = "resolvedEntity_saUtr"
+      when(mockEntityResolverConnector.resolveBy(anyString())(any[HeaderCarrier]()))
+        .thenReturn(
+          Future.successful(Entity(passedBackEntityId, saUtr = Some(resolvedEntity_saUtr), nino = None, itsa = None)))
+
+      when(mockEntityResolverConnector.update(any[Entity]())(any[HeaderCarrier]()))
+        .thenReturn(Future.successful(
+          Entity(passedBackEntityId, Some(resolvedEntity_saUtr), nino = None, itsa = Some(passedBackItsaId))))
+
+      when(mockEntityResolverConnector.update(any[Entity]())(any[HeaderCarrier]()))
+        .thenReturn(Future.successful(
+          Entity(passedBackEntityId, Some(resolvedEntity_saUtr), nino = None, itsa = Some(passedBackItsaId))))
+
+      val postData: JsValue = Json.parse(s"""{"entityId": "$passedBackEntityId", "itsaId": "$passedBackItsaId"}""")
+      val fakePostRequest = FakeRequest("POST", "", Headers("Content-Type" -> "application/json"), postData)
+      val response = controller.confirm().apply(fakePostRequest)
+      status(response) mustBe OK
+      contentAsString(response) mustBe """{"reason":"itsaId successfully linked to entityId"}"""
+    }
+
+    // Case 1.5
+    """return UNAUTHORIZED (401) when entityId already has a different itsaId linked to it in entity resolver""" in new TestSetup {
+      reset(mockEntityResolverConnector)
+      reset(mockAuthConnector)
```

Comment:
I appreciate this is probably not your code, but why after creating a new instance of `TestSetup` does anything need to be `reset` ?

Reply:
I preferred to stick to the test fixture strategy someone else wrote before me.

That someone created a _"mock collaborator"_ once, before all tests cases, and then shared it between multiple test cases. And so I did: just to keep the same pattern! The consequence is that I had to reset the state of the shared mocks  for each of the test cases and couldn't enable parallel executions.

To be fully honest, I'd rather prefer the adoptions of the _"fresh fixture"_ strategy (instead of _"shared fixture"_) ... but ... I was unsure I could change someone else code to accommodate mine.

Reply:
If it's wrong, fix it :)

Reply:
Ok, fine @spifix-hmrc !

Since you are encouraging me to apply some more improvements, I just pushed changes to unit tests that adopt the so-called  _"fresh fixture"_ setup strategy (see http://xunitpatterns.com/Fresh%20Fixture.html for details) which I believe to be more appropriate for our specific circumstances.